### DISCUSSION TITLE
BigDecimal generating a decimal point value

### DISCUSSION THREAD

**Discussion by YroslavNykolyshyn** [instancio/instancio#1170] [Q&A]
Hi everyone! 😊

I have a method that generates **BigDecimal** values with 3 digits after the decimal point, ensuring the last digit is not zero. Since I couldn't find a built-in method for this in the **BigDecimal** generation, I created a custom one:
```
instancioApi.generate(Select.fields().ofType(BigDecimal.class),  gen -> gen.text().pattern("#d#d#d.#d#d1").as(BigDecimal::new));
```
Is there a way to specify this pattern globally so it applies to all **BigDecimal** fields?

**Comment by armandino** [comment]
Hi @YroslavNykolyshyn! Yes, you can achieve this by implementing the `GeneratorProvider` SPI. Something like this should work I think:

```java
class GeneratorProviderImpl implements GeneratorProvider {

        @Override
        public GeneratorSpec<?> getGenerator(Node node, Generators gen) {
            if (node.getTargetClass() == BigDecimal.class) {
                return gen.text().pattern("#d#d#d.#d#d1").as(BigDecimal::new);
            }

            return null;
        }
    }
```

You can see a complete working example here: https://github.com/instancio/instancio-samples/tree/main/generator-provider-sample
